### PR TITLE
iss2 Terminology telegram bot

### PR DESCRIPTION
**Pull request by annkupriyanova**

### CODE REVIEW

File: terminology_bot/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (cdeler) [inline terminology_bot/Dockerfile:1]: It's look like you create your own container with pg based on the ubuntu. Is it necessary?

We might use something like https://hub.docker.com/_/postgres/

File: terminology_bot/bot.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+        self.dispatcher = self.updater.dispatcher
+        self.term_collection = TermCollection()
+        self.cur_term = {}
+
+    def start(self, bot, update):
+        """
+        Sends the greeting message with the start menu: 'Add new term' and 'Get list of terms' options
+        :return: the state START_MENU
+        """
+        update.message.reply_text('Hello! I am Terminology Bot. Send /cancel to stop talking to me.',
```
Comment (cdeler) [inline terminology_bot/bot.py]: I'm sure that good practise is to use i18n to have (in future) abitlity to translate our UI (https://docs.python.org/3/library/i18n.html, gettext)

Please create an issue to backlog to cover all UI-related strings by i18n calls.

File: terminology_bot/install.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+docker kill pg_terminology
+docker rm /pg_terminology
+docker build -t eg_postgresql .
+docker run -p 5555:5432 -d --name pg_terminology eg_postgresql
+sleep 10
```
Comment (cdeler) [inline terminology_bot/install.sh:5]: After this line this script become unclear for me. Could you please clarify the purpose of the script?

**Review by cdeler — CHANGES_REQUESTED** [review]
Please contact with @reason-for-risen 

We have to coordinate our high-level work scheme. I'd think that its bad idea to store the data, collected by TG bot, inside the DB (directly). I assume to pass this data to wiki-api.

But you are able to convice me for your point of view.

File: terminology_bot/install.sh (hunk 3 above)
Comment (annkupriyanova) [reply terminology_bot/install.sh:5]: The purpose of the script is to create a container with DB and then instantiate it with the schema (tables) and initial set of terms.

File: terminology_bot/Dockerfile (hunk 1 above)
Comment (annkupriyanova) [reply terminology_bot/Dockerfile:1]: As for storing data in DB, there was such a requirement from Rasul.

Comment (annkupriyanova) [reply terminology_bot/Dockerfile:1]: As for postgres official image, yes, we may use it.

File: terminology_bot/bot.py (hunk 2 above)
Comment (annkupriyanova) [reply terminology_bot/bot.py]: The issue #6 is created.

**Review by cdeler — APPROVED** [review]

File: terminology_bot/Dockerfile (hunk 1 above)
Comment (orasul) [reply terminology_bot/Dockerfile:1]: Definitely should use Postgres image.

File: terminology_bot/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+python_telegram_bot==11.1.0
+psycopg2==2.7.6.1
+SQLAlchemy==1.2.14
+telegram==0.0.1
```
Comment (orasul) [inline terminology_bot/requirements.txt]: telegram==0.0.1 - bot.py doesn't work with this line.
Fails with:
Traceback (most recent call last):
  File "bot.py", line 4, in <module>
    from telegram import (ReplyKeyboardMarkup, ReplyKeyboardRemove)
ImportError: cannot import name 'ReplyKeyboardMarkup'

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,8 @@
+terminology_bot/tmp
+venv
+.idea
+
+terminology_bot/config.mine.ini
```
Comment (orasul) [inline .gitignore:5]: Why terminology_bot/config.mine.ini is git ignored?

File: README.md [hunk 6]
```diff
@@ -1 +1,17 @@
...
+ 2. Execute *install.sh* script.
+
+    The script will:
+      - create Docker container with PostgreSQL,
+      - install the project dependencies from requirements.txt,
+      - execute *database.py* to create DB schema and insert initial data.
+
+ 3. Get your private token from Telegram BotFather (run */newbot* command and follow BotFather's instructions).
+ 4. Insert the *token* into the *config.ini* file.
+ 5. Set *multimedia_dir* in the *config.ini* file to store multimedia files for the terms.
```
Comment (orasul) [inline README.md]: "." is added to path specified in config.ini. Shouldn't be added.

File: terminology_bot/bot.py [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+
+    def image(self, bot, update):
+        """
+        Saves image of the current term to DB
+        """
+        user = update.message.from_user
+        photo_file = bot.get_file(update.message.photo[-1].file_id)
+
+        filename_sha1 = hashlib.sha1(bytes(f'image_{self.cur_term.name}', encoding='utf8')).hexdigest()
+        photo_file.download(f".{params['multimedia_dir']}/{filename_sha1}")
```
Comment (orasul) [inline terminology_bot/bot.py]: f".{params['multimedia_dir']}/{filename_sha1}" ==> f"{params['multimedia_dir']}/images/{filename_sha1}.jpg"

File: terminology_bot/bot.py [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+
+        logger.info('User %s uploaded the image for the term "%s"', user.first_name, self.cur_term.name)
+
+        self.term_collection.update(self.cur_term.id, {'image': f'image_{self.cur_term.name}'})
+
+        update.message.reply_text(_('Awesome!'), reply_markup=self.reply_menu_keyboard)
+        return self.CHOOSE_OPTION
+
+    def audio(self, bot, update):
```
Comment (orasul) [inline terminology_bot/bot.py]: didn't work

File: terminology_bot/bot.py [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+from telegram import (ReplyKeyboardMarkup, ReplyKeyboardRemove)
+from telegram.ext import (Updater, CommandHandler, MessageHandler, Filters, RegexHandler,
+                          ConversationHandler)
+
+from config import get_config
+from term_collection import TermCollection
+from database import POSEnum
+
+
+t = gettext.translation('terminology_bot', '/usr/share/locale')
```
Comment (orasul) [inline terminology_bot/bot.py]: Failed with:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/gettext.py", line 529, in translation
    raise OSError(ENOENT, 'No translation file found for domain', domain)
FileNotFoundError: [Errno 2] No translation file found for domain: 'terminology_bot'

**Review by orasul — CHANGES_REQUESTED** [review]
Need to fix:
1) requirements.txt
2) Dockerfile
3) multimedia files path
4) gettext.translation

File: terminology_bot/bot.py (hunk 9 above)
Comment (annkupriyanova) [reply terminology_bot/bot.py]: I've fixed i18n, added files with the translation for Russian. Commit https://github.com/annkupriyanova/wiki-framework/pull/5/commits/0d699cb7fbdca5b0ded6db50e406cf489dc8be51

File: terminology_bot/requirements.txt (hunk 4 above)
Comment (annkupriyanova) [reply terminology_bot/requirements.txt]: Deleted the line "telegram==0.0.1" in commit https://github.com/annkupriyanova/wiki-framework/commit/0d699cb7fbdca5b0ded6db50e406cf489dc8be51

File: .gitignore (hunk 5 above)
Comment (annkupriyanova) [reply .gitignore:5]: config.mine.ini file is git ignored because it's a local copy for config.ini with my own bot token.

File: README.md (hunk 6 above)
Comment (annkupriyanova) [reply README.md]: Commit https://github.com/annkupriyanova/wiki-framework/pull/5/commits/f0df298c702feeea64ddd1cdf2bf3bd1733650f1: "." was deleted. Now you should set an absolute path to *multimedia_dir* in *config.ini*

File: terminology_bot/bot.py (hunk 8 above)
Comment (annkupriyanova) [reply terminology_bot/bot.py]: @orasul, do you mean the problem of downloading multimedia files?

File: terminology_bot/bot.py (hunk 7 above)
Comment (annkupriyanova) [reply terminology_bot/bot.py]: Fixed in commit https://github.com/annkupriyanova/wiki-framework/pull/5/commits/fde248160994c48824d736f63a9bfee3b68896bf

File: terminology_bot/bot.py (hunk 8 above)
Comment (orasul) [reply terminology_bot/bot.py]: I've uploaded voice message and multimedia file wasn't placed in multimedia dir.

**Review by smetam — APPROVED** [review]

**Review by orasul — APPROVED** [review]

**Pull request merged**